### PR TITLE
feat: login

### PR DESCRIPTION
**Pull request by TomWoodward**

### CODE REVIEW

File: src/test/browserutils.ts [hunk 1]
```diff
@@ -117,5 +117,5 @@ export const checkLighthouse = async(target: puppeteer.Browser, urlPath: string)
   expect(lhr.categories.customAccessibility.score).toBeGreaterThanOrEqual(1);
   expect(lhr.categories.accessibility.score).toBeGreaterThanOrEqual(1);
   expect(lhr.categories.seo.score).toBeGreaterThanOrEqual(0.8);
-  expect(lhr.categories['best-practices'].score).toBeGreaterThanOrEqual(0.93);
+  expect(lhr.categories['best-practices'].score).toBeGreaterThanOrEqual(0.87);
```
Comment (philschatz) [inline src/test/browserutils.ts]: What caused the drop?

File: src/test/setup.ts [hunk 2]
```diff
@@ -21,6 +21,8 @@ expect.extend({
   toMatchImageSnapshot,
 });
 
+jest.mock('ally.js/style/focus-within');
```
Comment (philschatz) [inline src/test/setup.ts]: When is ally.js loaded in the real code? I probably missed it while reviewing.

**Review by philschatz — APPROVED** [review]
Why the switch from a down-caret to a hamburger menu when the screen is narrow?

Kapture 2019-05-06 at 16 20 17


The code looks fine aside from the other couple of comments in this review.

**Comment by TomWoodward** [comment]
@philschatz it changes to a hamburger on small screens because the menu works as an overlay on small screens and the hamburger is more intuitive for that, openstax.org does the same thing

File: src/test/browserutils.ts (hunk 1 above)
Comment (TomWoodward) [reply src/test/browserutils.ts]: the 403 from accounts indicating the user is not logged in, i couldn't find a way to make chrome ignore the error status in the request

```
Failed to load resource: the server responded with a status of 403 (Forbidden)
```

File: src/test/setup.ts (hunk 2 above)
Comment (philschatz) [reply src/test/setup.ts]: Thanks! And it seems to be to support [:focus-within in IE](https://developer.mozilla.org/en-US/docs/Web/CSS/:focus-within)

**Review by philschatz — APPROVED** [review]

**Pull request merged**